### PR TITLE
Cloudfront auth entry

### PR DESCRIPTION
**Pull request by uriahcarpenter**

### CODE REVIEW

File: _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        };
+        callback(null, response);
+    }
+
+    // Continue request processing if authentication passed
+    callback(null, request);
+};
+```
+
+This works surprising well; however, there was a lot of room for improvement. The most glaring obvious scalability problem is having a single shared password. One option we considered was to extend this code to improve the loading of passwords; the major downsides were still needing to manually manage user credentials and requiring users to remember yet another password.
```
Comment (bdotte) [inline _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md]: This works surprising**ly** well

File: _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+We use [Google's G Suite](https://gsuite.google.com/) internally for email; we thought if we were to leverage their support of [OpenID Connect](http://openid.net/connect/) as a relying party we could completely remove the need for our Lambda@Edge function to know anything about usernames or passwords.
```
Comment (bdotte) [inline _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md]: rel**a**ying party?

File: _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+At this point, the plan seemed clear:
+
+1. Configure CloudFront to inspect HTTP requests as they were received.
+2. If the request is not already __authenticated__, use one of the many [OpenID implementations](http://openid.net/developers/certified/) to redirect the user to the Google login UX.
+5. Perform user __authorization__ (email whitelist, Google Groups membership, etc.)
+3. Set a stateless JWT authentication token, as a cookie, with a configurable TTL.
+4. Redirect user to original request path.
+
+As what seems happen with every new project, the original plan never lasts long. Lambda@Edge has a few major [limitations](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/cloudfront-limits.html#limits-lambda-at-edge) that interfered:
```
Comment (bdotte) [inline _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md]: As is the case with every new project

File: _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md (hunk 2 above)
Comment (PaytonGarland) [reply _posts/2018-02-07-AWS-CloudFront-User-Authentication-using-Lambda@Edge.md]: [Relying Party](http://openid.net/connect/faq/) is "an app that outsources its user authentication function to an IDP"

**Pull request merged**